source/material_model/simple_nonlinear.cc

```diff
@@ -0,0 +1,371 @@
+/*
+  Copyright (C) 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/material_model/simple_nonlinear.h>
+#include <aspect/utilities.h>
+#include <aspect/newton.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+
+  namespace MaterialModel
+  {
+    template <int dim>
+    std::vector<double>
+    SimpleNonlinear<dim>::
+    compute_volume_fractions( const std::vector<double> &compositional_fields) const
```

Comment:
done: #1670 ;)
will be on my list after the newton is merged, or when I am tired of newton for a moment :)